src/cli.js

```diff
@@ -101,7 +129,8 @@ function Validator() {
       path.extname(filename)
     );
     const schemaLocation =
-      args.schema || (await getSchemaUrlForFilename(filename, cache, ttl));
+      args.schema || await getInlineSchemaUrl(filename) || (await getSchemaUrlForFilename(filename, cache, ttl));
```

Comment:
I'm not sure what is the priority you may prefer between inline or CLI arg.
Currently, the tool only allows one file to be checked, so the command line argument should be given priority in my opinion, but I saw that you were planning to manage directories or glob patterns one day.
So in this case, I think it might be debatable but it's probably outside the scope of this PR.